### ISSUE TITLE
Thief Deadly Shadow game and loading error with d3d8

### ISSUE THREAD

**Issue by fl82** [crosire/d3d8to9#126]
Hello,

I am trying to use the d3d8.dll as supposed to run already with previous patches with the old game Thief Deadly Shadow with the latest minimalistic UI and various patches for modern o.s. and when start with the latest d3d8.dll on the main game directory, at half the loading of the game resources it crash with an application error referring to an istruction 0x10dcae29 referring to the memory @ 0x00000010 and the memory couldn't be read. If I remove the dll, it loads the usual engine one that should be Dx8.1 and the game works even if not really fast.
The config is a netbook Atom with a GMA3600 gpu and Win 8.1 (Directx11 installed) 32bit. It's a Directx9.0c gpu but it's common to have compatibility problems with games.
I can't see any logs so if it exists some debug version I could try it.
I was thinking this patch might make the game working better considering it's a native Dx9 gpu and not very oriented for retrogaming.

Thanks

**Comment by Hexaae** [comment]
Am I wrong or an old crosire's d3d8to9 version is included in the Sneaky Upgrade patch for Thief 3 Deadly Shadow?

I've also found an incompatibility with "FPS Monitor" + d3d8to9 that makes Thief 3 crash at launch (!). Without FPS Monitor, it won't crash though, or when using "FPS Monitor" + d3d8.dll from https://github.com/ThirteenAG/d3d8-wrapper .
This means there's a conflict between this dll and "FPS Monitor" (or vice versa 😉 ) for some strange reason.

**Comment by elishacloud** [comment]
> Without FPS Monitor it won't crash though

This means it is likely not related to d3d8to9.

> when using "FPS Monitor" + d3d8.dll from https://github.com/ThirteenAG/d3d8-wrapper .

The dll from ThirteenAG's d3d8-wrapper has it's own FPS monitor built into it.

> I've also found an incompatibility with "[FPS Monitor](https://fpsmon.com/en/)" + d3d8to9 that makes Thief 3 crash at launch (!).

What happens if you only use Thief 3 and FPS Monitor without any other dll's or files?

**Comment by Hexaae** [comment]
> What happens if you only use Thief 3 and FPS Monitor without any other dll's or files?

FPSM overlay won't appear but at least game won't crash at launch

**Comment by elishacloud** [comment]
That is probably because FPSM does not work with Direct3D8.  This would also invalidate the test when using "FPS Monitor" + d3d8.dll from https://github.com/ThirteenAG/d3d8-wrapper, as ThirteenAG's d3d8-wrapper is also using Direct3D8.  It would appear, to me, that FPSM has a bug with this game, and you need to use d3d8to9 (to convert the game to Direct3D9) to even get FPSM to work with this game and expose the bug in it.

If you just want to see the FPS why not use [Intel's PresentMon](https://github.com/GameTechDev/PresentMon).  That should work fine with this game and d3d8to9.

**Comment by Hexaae** [comment]
Yes I don't think DX8 is supported by FPSM. The annoying thing is the crash. Would be ok if didn't work (i.e. no overlay) but did not crash. Let's see if FPSM will have a look why it crashes with wrappers like d3d8to9...

**Comment by elishacloud** [comment]
> Let's see if FPSM will have a look why it crashes with wrappers like d3d8to9...

Are you sure that FPSM has issues with wrappers?  Have you tried FPSM with other games and d3d8to9?  I guess it is just an issue with FPSM and Thief 3 Deadly Shadow.

**Comment by hammadnadeemx** [comment]
if your gpu supports vulkan try d8vk instead. I have tested it with thief deadly shadows and it works wells. https://github.com/AlpyneDreams/d8vk
